dbuf-lsp/src/core/ast_access/location.rs

```diff
@@ -0,0 +1,146 @@
+//! * LocationHelpers, helpers for Location type.
+//!
+
+use tower_lsp::lsp_types;
+use tower_lsp::lsp_types::Range;
+
+use dbuf_core::ast::parsed::location::{self, Offset};
+
+pub type Position = location::Offset;
+pub type Location = location::Location<Position>;
+
+/// Helpers for dbuf-core::Position type.
+pub trait PositionHelpers {
+    /// Creates new position.
+    fn new(lines: u32, columns: u32) -> Self;
+    /// Get line of position.
+    fn get_line(&self) -> u32;
+    /// Get column of position.
+    fn get_column(&self) -> u32;
+    /// Get column as mutable.
+    fn get_column_mut(&mut self) -> &mut usize;
+    /// Convers Position to lsp_types::Position;
+    fn to_lsp(&self) -> lsp_types::Position;
+}
+
+impl PositionHelpers for Position {
+    fn new(lines: u32, columns: u32) -> Self {
+        Position {
+            lines: lines as usize,
+            columns: columns as usize,
+        }
+    }
+    fn get_line(&self) -> u32 {
+        self.lines as u32
+    }
+    fn get_column(&self) -> u32 {
+        self.columns as u32
+    }
+    fn get_column_mut(&mut self) -> &mut usize {
+        &mut self.columns
+    }
+    fn to_lsp(&self) -> lsp_types::Position {
+        lsp_types::Position {
+            line: self.lines as u32,
+            character: self.columns as u32,
+        }
+    }
+}
+
+/// Helpers for dbuf-core::Location type.
+pub trait LocationHelpers {
+    /// Returns empty location. Typically ((0, 0), (0, 0))
+    fn new_empty() -> Self;
+    /// Returns location with start and end.
+    fn new(start: Position, end: Position) -> Self;
+    /// Convers Location to lsp_types::Range;
+    fn to_lsp(&self) -> Range;
+    /// Check if cursor position in location.
+    ///
+    /// If `p == self.end`, returns true, corresponding
+    /// to lsp_type::Range specification.
+    fn contains(&self, p: lsp_types::Position) -> bool;
+    /// Get start of location.
+    fn get_start(&self) -> Position;
+    /// Get end of location.
+    fn get_end(&self) -> Position;
+    /// Sets start of location.
+    fn set_start(&mut self, start: Position);
+    /// Sets end of location.
+    fn set_end(&mut self, end: Position);
+}
+
+impl LocationHelpers for Location {
+    fn new_empty() -> Self {
+        Self {
+            start: Position {
+                lines: 0,
+                columns: 0,
+            },
+            length: Position {
+                lines: 0,
+                columns: 0,
+            },
+        }
+    }
+
+    fn new(start: Position, end: Position) -> Self {
+        let mut ans = Self::new_empty();
+        ans.set_start(start);
+        ans.set_end(end);
+        ans
+    }
+
+    fn to_lsp(&self) -> Range {
+        Range {
+            start: self.start.to_lsp(),
+            end: self.end().to_lsp(),
+        }
+    }
+
+    fn contains(&self, p: lsp_types::Position) -> bool {
+        let line = p.line as usize;
+        let char = p.character as usize;
+
+        if self.start.lines > line {
```

Comment:
I feel like `Ord` instance for `Offset` should help here since it imposes lexicographic ordering which is precisely what's needed here